tasks/internal/push-artifacts-to-cdn-task/push-artifacts-to-cdn-task.yaml

```diff
@@ -881,18 +881,27 @@ spec:
             FILE=$(jq -c --arg i "$i" '.files[$i|tonumber]' <<< "$COMPONENT")
             SOURCE=$(jq -er '.source' <<< "$FILE") \
               || (echo "Missing files.source for component." && exit 1 )
-            BINARY=$(jq -er '.binary' <<< "$FILE")
-            case "${SOURCE}" in
-              (*darwin*)
-                tar -czf "${COMPRESSED_DIR}/${SOURCE}" -r "macos/${BINARY}"
-                ;;
-              (*linux*)
-                tar -czf "${COMPRESSED_DIR}/${SOURCE}" -r "linux/${BINARY}"
-                ;;
-              (*windows*)
-                zip -r "${COMPRESSED_DIR}/${SOURCE}" "windows/${BINARY}"
-                ;;
-            esac
+            MUST_COMPRESS="$(jq 'any(.delivery_format[]; . == "compressed")' <<< "$FILE")"
+            PUSH_RAW="$(jq 'any(.delivery_format[]; . == "raw")' <<< "$FILE")"
+
+            if [ "$MUST_COMPRESS" = "true" ]; then
+              case "${SOURCE}" in
+                (*darwin*)
+                  tar -czf "${COMPRESSED_DIR}/${SOURCE}" -r "${CONTENT_DIR}/macos/*"
+                  ;;
+                (*linux*)
+                  tar -czf "${COMPRESSED_DIR}/${SOURCE}" -r "${CONTENT_DIR}/linux/*"
+                  ;;
+                (*windows*)
+                  zip -r "${COMPRESSED_DIR}/${SOURCE}" "${CONTENT_DIR}/windows/*"
+                  ;;
+              esac
+            fi
+
+            # if "raw" is listed in the delivery_mode, make sure the binaries will be also pushed
+            if [ "$PUSH_RAW" = "true" ]; then
+              find "${CONTENT_DIR}/unsigned" "${CONTENT_DIR}/linux" -type f -exec cp "{}" "${COMPRESSED_DIR}/" \;
```

Comment:
Why only linux? Also, with this it's no longer suitable to call it COMPRESSED_DIR. Maybe change it to DELIVERY_DIR or something like that?